nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -535,6 +566,44 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   return transformed_plan;
 }
 
+double RegulatedPurePursuitController::dot(double v1[], double v2[])
+{
+  // Supports only 2D
+  return  (v1[0]*v2[0]) + (v1[1]*v2[1]);
+}
+
+double RegulatedPurePursuitController::checkOrientation(const geometry_msgs::msg::PoseStamped & pose )
+{
+  double dot_sign = 0.0;
+  double iter = 1;
+  double cusp_distance = 0.0;
+  bool not_in_cusp = true;
+
+  do
+  {
+    // We have two vectors for the dot product OA and AB. Determining the vectors.
+    double oa_x = global_plan_.poses[iter].pose.position.x - global_plan_.poses[iter-1].pose.position.x;
+    double oa_y = global_plan_.poses[iter].pose.position.y - global_plan_.poses[iter-1].pose.position.y;
+    double ab_x = global_plan_.poses[iter+1].pose.position.x - global_plan_.poses[iter].pose.position.x;
+    double ab_y = global_plan_.poses[iter+1].pose.position.y - global_plan_.poses[iter].pose.position.y;
+    double OA[] = {oa_x,oa_y};
```

Comment:
Never do this in C++, we have access to such amazing data structures and tools. Also, there's really not a strong reason to throw  these into an array since you're just doing math operations on them and then throwing them out. Just do your thing with the `_x` and `_y` variables, no need to combine them unless there's something else you're doing like passing them around a bunch or storing them longer term.